packages/@ama-sdk/core/src/fwk/param-deserialization.ts

```diff
@@ -0,0 +1,212 @@
+  return splitParamElements(serializedParamValue, paramSerialization);
+}
+
+/**
+ * Deserialize path parameters of type object
+ * OpenAPI Parameter Serialization {@link https://swagger.io/specification | documentation}
+ * @param serializedParamValue serialized path parameter value
+ * @param paramSerialization parameter serialization
+ */
+function deserializeObjectPathParams(serializedParamValue: string, paramSerialization: ParamSerialization) {
+  // NOTE: The splitParamElements function is called since object path parameters can be split by delimiters based on the serialization style
+  // NOTE: There is an exception for path parameters of exploded style 'matrix' which is not serialized like its corresponding array
+  // This exception is serialized like this (prefixed by a ';'): ';prop1=value1;prop2=value2'
+  const splitObject: string[] = paramSerialization.style === 'matrix' && paramSerialization.explode
+    ? serializedParamValue.substring(1).split(';').map((value) => decodeURIComponent(value))
+    : splitParamElements(serializedParamValue, paramSerialization)!;
+
+  // NOTE: Object path parameters that are exploded are serialized as 'prop=value'
+  if (paramSerialization.explode) {
+    return splitObject.reduce((obj: { [key: string]: PrimitiveType }, serializedProperty) => {
+      const [key, value] = serializedProperty.split('=').map((v) => decodeURIComponent(v));
+      obj[key] = value;
+      return obj;
+    }, {} as { [key: string]: PrimitiveType });
+  }
+
+  return objectFromPairwiseArray(splitObject);
+}
+
+/**
+ * Deserialize path parameters based on the values of exploded and style and the parameter type
+ * OpenAPI Parameter Serialization {@link https://swagger.io/specification | documentation}
+ * @param serializedPathParams serialized path parameters
+ * @param pathParamSerialization path parameter serialization
+ */
+export function deserializePathParams<T extends { [key: string]: string }>(
+  serializedPathParams: T,
+  pathParamSerialization: { [p in keyof T]: ParamSerialization & { paramType: ParamTypeForDeserialization } }
+): { [p in keyof T]: SupportedParamType } {
+  return Object.entries(serializedPathParams).reduce((acc, [pathParamName, serializedParamValue]) => {
+    const paramSerialization = pathParamSerialization[pathParamName];
+    let deserializedValue: SupportedParamType;
+    if (paramSerialization.paramType === 'array') {
+      deserializedValue = deserializeArrayPathParams(serializedParamValue, paramSerialization);
+    } else if (paramSerialization.paramType === 'object') {
+      deserializedValue = deserializeObjectPathParams(serializedParamValue, paramSerialization);
+    } else {
+      deserializedValue = deserializePrimitivePathParams(serializedParamValue, paramSerialization);
+    }
+    if (deserializedValue) {
+      acc[pathParamName as keyof T] = deserializedValue;
+    }
+    return acc;
+  }, {} as { [p in keyof T]: SupportedParamType });
```

Comment:
Same comment as above:
```suggestion
  }, {} as Record<string, SupportedParamType>) as { [p in keyof T]: SupportedParamType };
```